Brokerages/Fxcm/FxcmBrokerage.cs

```diff
@@ -0,0 +1,364 @@
+
+            return _openOrders.Values.ToList().Where(x => OrderIsOpen(x.getFXCMOrdStatus().getCode())).Select(ConvertOrder).ToList();
+        }
+
+        /// <summary>
+        /// Gets all holdings for the account
+        /// </summary>
+        /// <returns>The current holdings from the account</returns>
+        public override List<Holding> GetAccountHoldings()
+        {
+            Log.TraceFormat("FxcmBrokerage.GetAccountHoldings()");
+
+            var holdings = _openPositions.Values.Select(ConvertHolding).ToList();
+
+            // Set MarketPrice in each Holding
+            var symbols = holdings.Select(x => ConvertSymbolToFxcmSymbol(x.Symbol)).ToList();
+            if (symbols.Count > 0)
+            {
+                var quotes = GetQuotes(symbols).ToDictionary(x => x.getInstrument().getSymbol());
+                foreach (var holding in holdings)
+                {
+                    MarketDataSnapshot quote;
+                    if (quotes.TryGetValue(ConvertSymbolToFxcmSymbol(holding.Symbol), out quote))
+                    {
+                        holding.MarketPrice = Convert.ToDecimal((quote.getBidClose() + quote.getAskClose()) / 2);
+                    }
+                }
+            }
+
+            return holdings;
+        }
+
+        /// <summary>
+        /// Gets the current cash balance for each currency held in the brokerage account
+        /// </summary>
+        /// <returns>The current cash balance for each currency available for trading</returns>
+        public override List<Cash> GetCashBalance()
+        {
+            Log.TraceFormat("FxcmBrokerage.GetCashBalance()");
+
+            return _accounts.Values.Select(account => 
+                new Cash(_fxcmAccountCurrency, Convert.ToDecimal(account.getCashOutstanding()), GetUsdConversion(_fxcmAccountCurrency))).ToList();
+        }
+
+        /// <summary>
+        /// Places a new order and assigns a new broker ID to the order
+        /// </summary>
+        /// <param name="order">The order to be placed</param>
+        /// <returns>True if the request for a new order has been placed, false otherwise</returns>
+        public override bool PlaceOrder(Order order)
+        {
+            Log.TraceFormat("FxcmBrokerage.PlaceOrder(): {0}", order);
+
+            if (!IsConnected)
+                throw new InvalidOperationException("FxcmBrokerage.PlaceOrder(): Unable to place order while not connected.");
+
+            if (order.Direction != OrderDirection.Buy && order.Direction != OrderDirection.Sell)
+                throw new ArgumentException("FxcmBrokerage.PlaceOrder(): Invalid Order Direction");
+
+            var symbol = ConvertSymbolToFxcmSymbol(order.Symbol);
+            var orderSide = order.Direction == OrderDirection.Buy ? SideFactory.BUY : SideFactory.SELL;
+            var quantity = (double)order.AbsoluteQuantity;
+            var accountId = _accounts.Keys.First();
```

Comment:
I understand Fxcm multiple accounts are used primarily for managed accounts: [PAMM](https://en.wikipedia.org/wiki/Percent_allocation_management_module) or LAMM.